### ISSUE TITLE
Should we have typscript alongside javascript examples?

### ISSUE THREAD

**Issue by svrnm** [open-telemetry/opentelemetry.io#1769]
@open-telemetry/docs-approvers , @open-telemetry/javascript-approvers:

The Elixir/Erlang docs uses Tabs to provide examples in Erlang&Elixir side-by-side ([see here](https://opentelemetry.io/docs/instrumentation/erlang/instrumentation/)), I was wondering if we want to do the same for JavaScript & TypeScript? 

This could be a nice set of **good first issues** / **help wanted** we can put out there

**Comment by cartermp** [comment]
How affected by ESM support would these samples be? I know that I tried standardizing on ESM for the JS examples because vscode will suggest converting `require` into `import` declarations. But for some reason I couldn't get instrumentation to load properly.

**Comment by dyladan** [comment]
I strongly support this. Typescript is what OTel itself is written in and for many early adopters it is the default choice for JS runtimes.

>How affected by ESM support would these samples be?

Not at all. ESM support is orthogonal to typescript support. Typescript support in OTel JS is very good.

> I know that I tried standardizing on ESM for the JS examples because vscode will suggest converting require into import declarations. But for some reason I couldn't get instrumentation to load properly.

autoinstrumentation does not support esm due to limitations in the nodejs module loader mechanism. those are being worked through but we're basically waiting on nodejs at this point.

**Comment by cartermp** [comment]
Awesome, that sounds fantastic. 100% in favor of adding typescript in then, and furthermore, I'd even love it if it was the default tab.

**Comment by legendecas** [comment]
This can be awesome! I believe TS as the default tab would be favorable as it can provide clearer type annotations on certain code snippets.

Regarding ESM, I don't think we are hasty evangelizing ESM usage as we don't have an ESM instrumentation released yet.

**Comment by svrnm** [comment]
OK, then let me raise a new ticket laying out what needs to be done. Thanks everyone :)
